docs/src/tutorials/Nonlinear programs/space_shuttle_reentry_trajectory.jl

```diff
@@ -0,0 +1,312 @@
+#       \rho_0 & = 0.002378,                                    \qquad & b_2 & =  0.621408 \times 10^{-3}, \\
+#          h_r & = 23800,                                       \qquad & q_r & =  17700 \sqrt{\rho} (0.0001 v)^{3.07}, \\
+#          c_L & = a_0 + a_1 \hat{\alpha},                      \qquad & q_a & =  c_0 + c_1 \hat{\alpha} + c_2 \hat{\alpha}^2 + c_3 \hat{\alpha}^3, \\
+#          c_D & = b_0 + b_1 \hat{\alpha} + b_2 \hat{\alpha}^2, \qquad & c_0 & =  1.0672181, \\
+# \hat{\alpha} & = 180 \alpha / \pi,                            \qquad & c_1 & = -0.19213774 \times 10^{-1}, \\
+#          R_e & = 20902900,                                    \qquad & c_2 & =  0.21286289 \times 10^{-3}, \\
+#            S & = 2690,                                        \qquad & c_3 & = -0.10117249 \times 10^{-5}.
+# \end{aligned}
+# ```
+
+# The reentry trajectory begins at an altitude where the aerodynamic forces are quite small
+# with a weight of $w = 203000$ (lb) and mass $m = w / g_0$ (slug), where $g_0 = 32.174$ (ft/sec$^2$).
+# The initial conditions are as follows:
+# 
+# ```math
+# \begin{aligned}
+#      h & = 260000 \text{ ft},  \qquad & v      & = 25600 \text{ ft/sec}, \\
+#   \phi & = 0      \text{ deg}, \qquad & \gamma & = -1    \text{ deg},    \\
+# \theta & = 0      \text{ deg}, \qquad & \psi   & = 90    \text{ deg}.
+# \end{aligned}
+# ```
+
+# The final point on the reentry trajectory occurs at the unknown (free) time $t_F$,
+# at the so-called terminal area energy management (TAEM) interface, which is defined by the conditions
+# 
+# ```math
+# h = 80000 \text{ ft}, \qquad v = 2500 \text{ ft/sec}, \qquad \gamma = -5 \text{ deg}.
+# ```
+
+# As explained in the book, our goal is to maximize the final crossrange,
+# which is equivalent to maximizing the final latitude of the vehicle,
+# i.e., $J = \theta(t_F)$.
+
+# ![Max crossrange shuttle reentry](https://i.imgur.com/1tUN3wM.png)
+
+# ### Approach 
+# We will use a discretized model of time, with a fixed number of discretized points, $n$.
+# The decision variables at each point are going to be the state of the vehicle and the controls commanded to it.
+# In addition, we will also make each time step size $\Delta t$ a decision variable;
+# that way, we can either fix the time step size easily,
+# or allow the solver to fine-tune the duration between each adjacent pair of points.
+# Finally, in order to approximate the derivatives of the problem dynamics, we will use either rectangular
+# or [trapezoidal](http://en.wikipedia.org/wiki/Trapezoidal_rule) integration.
+
+# > #### Disclaimer
+# > Do not try to actually land a Space Shuttle using this notebook! 😛
+# > There's no mesh refinement going on, which can lead to unrealistic trajectories having position
+# > and velocity errors with orders of magnitude $10^4$ ft and $10^2$ ft/sec, respectively.
+
+using Interpolations
+using JuMP, Ipopt #, KNITRO
+
+## Global variables
+w  = 203000.0  # weight (lb)
```

Comment:
Do you have the stack trace of the error from setting these to `const`? We should fix the root cause.